components/controlled_dialog.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from "react";
+
+import { Button, IButtonProps, Intent } from "@blueprintjs/core";
+import { CardActions } from "df/components/card";
+import { Dialog } from "df/components/dialog";
+
+export interface IProps {
+  title?: string;
+  ref?: React.RefObject<ControlledDialog>;
+  button?: React.ReactElement<any>;
+  onConfirm?: () => void;
+  onClose?: (e: any) => void;
+  confirmButtonProps?: IButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>;
+  cancelButtonText?: string;
+  confirmHidden?: boolean;
+  ignore?: boolean;
+  width?: string;
+}
+
+interface IState {
+  isOpen: boolean;
+}
+
+export class ControlledDialog extends React.Component<IProps, IState> {
```

Comment:
Hooks now plz! This is an easy one :)

Reply:
I was hoping you wouldn't say this, it's not easy because the files_list component attaches a ref to this component, and then uses that ref to call the showDialog function on this class, which sets state inside the class. It's gross but I didn't want to fix it. I guess I can fix it though.